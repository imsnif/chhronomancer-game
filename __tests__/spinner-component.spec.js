import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

test('Spinner => renders correctly', () => {
  jest.useFakeTimers()
  const Spinner = require('../components/spinner').default
  const tree = renderer.create(
    <Spinner />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Spinner => renders correctly (first spin)', () => {
  jest.useFakeTimers()
  const Spinner = require('../components/spinner').default
  const tree = renderer.create(
    <Spinner />
  )
  jest.runTimersToTime(50)
  expect(tree.toJSON()).toMatchSnapshot()
})

test('Spinner => renders correctly (second spin)', () => {
  jest.useFakeTimers()
  const Spinner = require('../components/spinner').default
  const tree = renderer.create(
    <Spinner />
  )
  jest.runTimersToTime(100)
  expect(tree.toJSON()).toMatchSnapshot()
})

test('Spinner => renders correctly (third spin)', () => {
  jest.useFakeTimers()
  const Spinner = require('../components/spinner').default
  const tree = renderer.create(
    <Spinner />
  )
  jest.runTimersToTime(150)
  expect(tree.toJSON()).toMatchSnapshot()
})

test('Spinner => renders correctly (fourth spin)', () => {
  jest.useFakeTimers()
  const Spinner = require('../components/spinner').default
  const tree = renderer.create(
    <Spinner />
  )
  jest.runTimersToTime(250)
  expect(tree.toJSON()).toMatchSnapshot()
})

test('Spinner => renders correctly (fifth spin)', () => {
  jest.useFakeTimers()
  const Spinner = require('../components/spinner').default
  const tree = renderer.create(
    <Spinner />
  )
  jest.runTimersToTime(300)
  expect(tree.toJSON()).toMatchSnapshot()
})

test('Spinner => renders correctly (sixth spin)', () => {
  jest.useFakeTimers()
  const Spinner = require('../components/spinner').default
  const tree = renderer.create(
    <Spinner />
  )
  jest.runTimersToTime(350)
  expect(tree.toJSON()).toMatchSnapshot()
})

test('Spinner => renders correctly (seventh spin)', () => {
  jest.useFakeTimers()
  const Spinner = require('../components/spinner').default
  const tree = renderer.create(
    <Spinner />
  )
  jest.runTimersToTime(400)
  expect(tree.toJSON()).toMatchSnapshot()
})

test('Spinner => renders correctly (eighth spin)', () => {
  jest.useFakeTimers()
  const Spinner = require('../components/spinner').default
  const tree = renderer.create(
    <Spinner />
  )
  jest.runTimersToTime(450)
  expect(tree.toJSON()).toMatchSnapshot()
})
