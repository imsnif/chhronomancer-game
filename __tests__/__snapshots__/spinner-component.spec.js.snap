// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner => renders correctly (eighth spin) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#329932",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#141313",
        "flex": 1,
        "fontFamily": "telegrama_raw",
        "textAlign": "center",
      }
    }
  >
    /
  </Text>
</View>
`;

exports[`Spinner => renders correctly (fifth spin) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#329932",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#141313",
        "flex": 1,
        "fontFamily": "telegrama_raw",
        "textAlign": "center",
      }
    }
  >
    -
  </Text>
</View>
`;

exports[`Spinner => renders correctly (first spin) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#329932",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#141313",
        "flex": 1,
        "fontFamily": "telegrama_raw",
        "textAlign": "center",
      }
    }
  >
    /
  </Text>
</View>
`;

exports[`Spinner => renders correctly (fourth spin) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#329932",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#141313",
        "flex": 1,
        "fontFamily": "telegrama_raw",
        "textAlign": "center",
      }
    }
  >
    /
  </Text>
</View>
`;

exports[`Spinner => renders correctly (second spin) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#329932",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#141313",
        "flex": 1,
        "fontFamily": "telegrama_raw",
        "textAlign": "center",
      }
    }
  >
    -
  </Text>
</View>
`;

exports[`Spinner => renders correctly (seventh spin) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#329932",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#141313",
        "flex": 1,
        "fontFamily": "telegrama_raw",
        "textAlign": "center",
      }
    }
  >
    |
  </Text>
</View>
`;

exports[`Spinner => renders correctly (sixth spin) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#329932",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#141313",
        "flex": 1,
        "fontFamily": "telegrama_raw",
        "textAlign": "center",
      }
    }
  >
    \\
  </Text>
</View>
`;

exports[`Spinner => renders correctly (third spin) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#329932",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#141313",
        "flex": 1,
        "fontFamily": "telegrama_raw",
        "textAlign": "center",
      }
    }
  >
    \\
  </Text>
</View>
`;

exports[`Spinner => renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#329932",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#141313",
        "flex": 1,
        "fontFamily": "telegrama_raw",
        "textAlign": "center",
      }
    }
  >
    |
  </Text>
</View>
`;
